lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -391,10 +407,9 @@ void StagesSubject::onStageInvalidate(const MayaUsdProxyStageInvalidateNotice& n
 #ifdef UFE_V2_FEATURES_AVAILABLE
     auto p = notice.GetProxyShape().ufePath();
     if (!p.empty()) {
-        Ufe::SceneItem::Ptr sceneItem = Ufe::Hierarchy::createItem(p);
-        if (sceneItem) {
-            Ufe::Scene::instance().notify(Ufe::SubtreeInvalidate(sceneItem));
-        }
+        // We can't send notification to clients from dirty propagation.
+        // Delay it till the new stage is actually set during compute.
+        fInvalidStages.insert(p);
     }
```

Comment:
I see you didn't put fInvalidStages inside the ifdef V2 block. So I think that either all the code that uses this var should be inside an ifdef or remove the ifdef from this block. The block above should be ifdef protected because subtreeinvalidate is V2 only. But then line 396 should be moved outside the block.

Reply:
It was mostly to leave the header file cleaner since code that is not guarded has no real performance impact (very low-frequency operation on empty set). It also builds with all flavors of the build, because the entire `ufe` folder is not included when building without UFE.

Let me know if you feel strongly about it and I can add the guards.

Reply:
My one concern is that the population of fInvalidStages is done inside an ifdef V2 block. So if someone later adds code to use this fInvalidStages var and that code is not inside a V2 block, the new code will not work since the set will never have anything in it.

Reply:
Addressed in ea81f449b09455330246e2d5a90cd918ef66b281